infrastructure/matrix/mautrix_client.go

```diff
@@ -36,15 +39,34 @@ type client struct {
 }
 
 func NewMautrixClient(homeserverURL string, enableListeners bool) (*client, error) {
-	mautrixClient, err := mautrix.NewClient(homeserverURL, "", "")
+	hsURL, err := url.Parse(homeserverURL)
 	if err != nil {
 		return nil, err
 	}
+	if hsURL.Scheme == "" {
+		hsURL.Scheme = "https"
+	}
 
-	var syncer mautrixSyncer
+	var syncer *mautrix.DefaultSyncer
 	if enableListeners {
-		syncer := mautrix.NewDefaultSyncer()
-		mautrixClient.Syncer = syncer
+		syncer = mautrix.NewDefaultSyncer()
+	} else {
+		// TODO: stub syncer?
+	}
+
+	mautrixClient := mautrix.Client{
```

Comment:
We should `&mautrix.Client`  right at this point, since we need to pass a pointer in client struct and then do the same for client struct literal since we intend to return pointer of that in the function.

Reply:
Also, I quite didn't understand from this PR what do we get out of creating `mautrix.Client{}` ourselves instead of `mautrix.NewClient()`? Perhaps you are aiming for something in a future PR?

Reply:
The problem is that `mautrix.NewClient()` creates an instance of `DefaultSyncer`, which makes it hard to mock in tests. Previously we were "hacking it" by creating our own instance and overriding the default one, but that wasn't working correctly, so I went ahead and stopped using the `mautrix.NewClient()` altogether.